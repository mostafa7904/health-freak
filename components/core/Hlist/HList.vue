<template>
  <component :is="tag">
    <template v-if="items.length">
      <!-- <slot v-for="(item, key) in items" :key="key" name="item"></slot> -->
    </template>
    <slot v-else></slot>
  </component>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'HealthFreakList',
  props: {
    items: {
      type: Array,
      default: () => [],
      validator: v => !!v
    },
    tag: {
      type: String,
      default: 'div'
    }
  },
  data: () => ({})
})
</script>
