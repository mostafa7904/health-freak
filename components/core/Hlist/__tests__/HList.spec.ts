// Libraries
import Vue from 'vue'

// Utilities
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

// Components
import HList from '../HList.vue'

describe('HList.vue', () => {
  let mountFunction: (options?: object) => Wrapper<Vue>

  let localVue: typeof Vue

  beforeEach(() => {
    localVue = createLocalVue()

    mountFunction = (options = {}) => {
      return mount(HList, {
        localVue,
        ...options
      })
    }
  })

  it('should render component and match snapshot', () => {
    const html = mountFunction().html()
    expect(html).toMatchSnapshot()
  })
})
