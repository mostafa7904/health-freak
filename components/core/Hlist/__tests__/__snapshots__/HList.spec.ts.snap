// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HList.vue should render component and match snapshot 1`] = `"<div></div>"`;
